django/contrib/staticfiles/storage.py

```diff
@@ -196,7 +196,9 @@ def converter(matchobj):
 
             if url_path.startswith('/'):
                 # Otherwise the condition above would have returned prematurely.
-                assert url_path.startswith(settings.STATIC_URL)
+                if not url_path.startswith(settings.STATIC_URL):
+                    raise ValueError("The URL path does not resolve to a valid static file.")
```

Comment:
This is a check in an undocumented, private API, so I'd not change this, see [comment](https://code.djangoproject.com/ticket/32508#comment:3). It's not user-facing.